src/estimagic/benchmarking/benchmark_reports.py

```diff
@@ -0,0 +1,171 @@
+            solution value.
+        results (dict): estimagic benchmarking results dictionary. Keys are
+            tuples of the form (problem, algorithm), values are dictionaries of the
+            collected information on the benchmark run, including 'criterion_history'
+            and 'time_history'.
+        runtime_measure (str): "n_evaluations", "n_batches" or "walltime".
+            This is the runtime until the desired convergence was reached by an
+            algorithm. This is called performance measure by Moré and Wild (2009).
+            Default is "n_evaluations".
+        normalize_runtime (bool): If True the runtime each algorithm needed for each
+            problem is scaled by the time the fastest algorithm needed. If True, the
+            resulting plot is what Moré and Wild (2009) called data profiles.
+            Default is False.
+        stopping_criterion (str): one of "x_and_y", "x_or_y", "x", "y". Determines
+            how convergence is determined from the two precisions.
+        x_precision (float or None): how close an algorithm must have gotten to the
+            true parameter values (as percent of the Euclidean distance between start
+            and solution parameters) before the criterion for clipping and convergence
+            is fulfilled. Default is 1e-4.
+        y_precision (float or None): how close an algorithm must have gotten to the
+            true criterion values (as percent of the distance between start
+            and solution criterion value) before the criterion for clipping and
+            convergence is fulfilled. Default is 1e-4.
+
+    Returns:
+        pandas.DataFrame: columns are the algorithms, indexes are the problems.
+            The values are the ranks of the algorithms for each problem,
+            0 means algorithms was the fastest, 1 means it was the second fastest
+            and so on. If an algorithm did not converge on a problem, the value is
+            "failed". If an algorithm did encounter an error during optimization,
+            the value is "error".
+
+    """
+    histories, converged_info = process_benchmark_results(
+        problems=problems,
+        results=results,
+        stopping_criterion=stopping_criterion,
+        x_precision=x_precision,
+        y_precision=y_precision,
+    )
+    scenarios = list({algo[1] for algo in results.keys()})
+
+    success_info = converged_info.replace({True: "success", False: "failed"})
+    for key, value in results.items():
+        if isinstance(value["solution"], str):
+            success_info.at[key] = "error"
+
+    solution_times = histories.groupby(["problem", "algorithm"])[runtime_measure].max()
+
+    if normalize_runtime:
+        solution_times = solution_times.unstack()
+        solution_times = solution_times.divide(solution_times.min(axis=1), axis=0)
+        solution_times = solution_times.stack(dropna=False)
+        solution_times.name = runtime_measure
```

Comment:
Since the normalization should not matter for the rank, I think this can go.

Reply:
Thanks! Yes, we don't actually need it here.